### PR TITLE
Extended SDK Metrics

### PR DESCRIPTION
**Pull request by sighphyre**
This adds an extended metrics format to the metrics ingested by Unleash and sent by running SDKs in the wild. Notably, we don't store this information anywhere new in this PR, this is just streamed out to Victoria metrics - the point of this project is insight, not analysis.

Two things to look out for in this PR:

- I've chosen to take extend the registration event and also send that when we receive metrics. This means that the new data is received on startup and on heartbeat. This takes us in the direction of collapsing these two calls into one at a later point
- I've wrapped the existing metrics events in some "type safety", it ain't much because we have 0 type safety on the event emitter so this also has some if checks that look funny in TS that actually check if the data shape is correct. Existing tests that check this are more or less preserved

### CODE REVIEW

File: src/lib/metrics.ts [hunk 1]
```diff
@@ -784,15 +792,36 @@ export default class MetricsMonitor {
...
+                        heartbeatEvent.metadata?.yggdrasilVersion ?? 'not-sent',
+                    specVersion:
+                        heartbeatEvent.metadata?.specVersion ?? 'not-sent',
+                });
+            } else {
+                clientSdkVersionUsage.increment({
+                    sdk_name: heartbeatEvent.sdkName,
+                    sdk_version: heartbeatEvent.sdkVersion,
+                    platformName: 'not-sent',
+                    platformVersion: 'not-sent',
```
Comment (sighphyre) [inline src/lib/metrics.ts:818]: @gardleopard I'm not sure this is the right way to deal with metrics we never get?

File: src/lib/metrics.ts [hunk 2]
```diff
@@ -784,15 +792,36 @@ export default class MetricsMonitor {
...
+            }
+
+            if (flagResolver.isEnabled('extendedMetrics')) {
                 clientSdkVersionUsage.increment({
-                    sdk_name: sdkName,
-                    sdk_version: sdkVersion,
+                    sdk_name: heartbeatEvent.sdkName,
+                    sdk_version: heartbeatEvent.sdkVersion,
+                    platformName:
+                        heartbeatEvent.metadata?.platformName ?? 'not-sent',
```
[the remark this answers is not in the document]
Comment (gardleopard) [reply src/lib/metrics.ts:805]: actually n/a is what we have used before

[the remark this answers is not in the document]
Comment (gardleopard) [reply src/lib/metrics.ts:805]: but that doesnt make sense. This not-sent might be better for this. maybe not-set is more accurate, but yeah use not set or not-sent. We know that it is not set, so unknown is bad

File: src/lib/metrics.ts (hunk 1 above)
Comment (gardleopard) [reply src/lib/metrics.ts:818]: maybe not-sent or not-set should be something else here like ignored?

**Review by gardleopard — APPROVED** [review]
LGTM

**Pull request merged**